File: RedesII/practica1/code/www/scripts/saludo.py
############################################################################
# saludo.py
#
# Script que recibe un nombre y responde con la cadena "Hola <nombre>!"
#
# Autores : Ricardo Riol Gonzalez y Alejandro Sanchez Sanz
############################################################################


#!/usr/bin/python
import sys 

dirFichero = sys.argv[2]
fichero = open(dirFichero, 'w')
if len(sys.argv) != 3:
	fichero.write("Error, se debe introducir dos un argumento de entrada\n")
else:
	campos = []
	campos = sys.argv[1].split('=')
	if (len(campos) != 2) or (campos[0] != "nombre"):
		fichero.write("Error, mal introducidos los argumentos: nombre=valor\n")
	else:
		fichero.write("Hola")
		separados = campos[1].split("+")
		for palabra in separados:
			fichero.write(" " + palabra);
		fichero.write("!\n")
fichero.close()

File: RedesII/practica1/www/scripts/saludo.php
#!/usr/bin/php
<?php
/***************************************************************************
* saludo.php
*
* Script que recibe un nombre y responde con la cadena "Hola <nombre>!"
*
* Autores : Ricardo Riol Gonzalez y Alejandro Sanchez Sanz
***************************************************************************/
$file = fopen($argv[2], "w");
if ($argc != 3) {
	fwrite($file, "Error, se deben introducir solo dos argumento de entrada". PHP_EOL);
} else {
	$campos = explode("=", $argv[1]);
	if ((count($campos) != 2) || (strcmp($campos[0], "nombre"))) {
		fwrite($file, "Error, mal introducidos los argumentos: nombre=valor". PHP_EOL);
	} else {
		fwrite($file, "Hola");
		$separados = explode("+", $campos[1]);
		for ($i=0, $num = count($separados); $i < $num; $i++) {
			fwrite($file, " ". $separados[$i]);
		}
		fwrite($file, "!". PHP_EOL);
	}
}
fclose($file);
?>

